### PR TITLE
Show all teachercon workshops in detail view

### PR DESCRIPTION
**Pull request by Hamms**
Create a new API route for retrieving all upcoming teachercon workshops (without authorization, completely public), and update the Summer Workshop Assignment loader to hit both that _and_ the route for retrieving "my" workshops.

### CODE REVIEW

File: dashboard/app/controllers/api/v1/pd/workshops_controller.rb [hunk 1]
```diff
@@ -60,6 +60,17 @@ def filter
     render json: {error: e.message}, status: :bad_request
   end
 
+  # GET /api/v1/pd/workshops/upcoming_teachercon
+  def upcoming_teachercon
+    workshops = Pd::Workshop.scheduled_start_on_or_after(Date.today.beginning_of_day).where(
```
Comment (mehalshah) [inline dashboard/app/controllers/api/v1/pd/workshops_controller.rb]: As per discussion, make sure this only includes the upcoming teachercons associated with this regional partner

File: dashboard/test/controllers/api/v1/pd/workshops_controller_test.rb [hunk 2]
```diff
@@ -565,6 +565,48 @@ class Api::V1::Pd::WorkshopsControllerTest < ::ActionController::TestCase
     assert_equal expected_workshops.map(&:id).sort, response.map {|x| x['id']}.sort
   end
 
+  test 'Teachercon workshops are public' do
```
Comment (mehalshah) [inline dashboard/test/controllers/api/v1/pd/workshops_controller_test.rb]: Augment this test to make sure that it only retrieves the teachercons this regional partner is associated with

**Comment by Hamms** [comment]
PTAL

**Comment by Hamms** [comment]
[Partners can't see an assigned TC summer workshop](https://trello.com/c/yKXyRJpY/391-partners-cant-see-an-assigned-tc-summer-workshop)

File: dashboard/app/controllers/api/v1/pd/workshops_controller.rb [hunk 3]
```diff
@@ -60,6 +61,43 @@ def filter
     render json: {error: e.message}, status: :bad_request
   end
 
+  # GET /api/v1/pd/workshops/upcoming_teachercon
+  # returns all upcoming teachercons for admins, associated teachercons for
+  # regional partners, and an empty set for everyone else.
+  def upcoming_teachercon
```
Comment (mehalshah) [inline dashboard/app/controllers/api/v1/pd/workshops_controller.rb]: nit - this should be `upcoming_teachercons`

Comment (Hamms) [reply dashboard/app/controllers/api/v1/pd/workshops_controller.rb]: :+1:

**Pull request merged**